skimage/measure/_find_contours.py

```diff
@@ -7,13 +9,14 @@
 _param_options = ('high', 'low')
 
 
-def find_contours(image, level=None,
-                  fully_connected='low', positive_orientation='low',
+def find_contours(image: np.typing.NDArray[np.float32], level: Optional[float] = None,
```

Comment:
Unfortunately, I think it will be harder to capture what we currently accept as an image. @lagru and @jni are working on figuring out what that annotation should look like.

Reply:
Let me know how I can help. Not sure how much more precise the type hint can be until [PEP 646](https://peps.python.org/pep-0646/) is available in all supported versions.

Reply:
For one, images are typically not limited to float64, although that is more and more becoming the format in practice.

Looks like PEP646 lands in 3.11, but I vaguely recall that there are workarounds for earlier versions.

Reply:
This seems relevant in the PEP:

> Gradual typing is enabled by the fact that unparameterised variadic classes are compatible with an arbitrary number of type parameters. This means that if existing classes are made generic, a) all existing (unparameterised) uses of the class will still work, and b) parameterised and unparameterised versions of the class can be used together (relevant if, for example, library code is updated to use parameters while user code is not, or vice-versa).

Also, agreed that images are not just float images in general.

Anyway, we don't have a concrete plan right now but we *do* want to use either NewType or Annotated to distinguish between "image" arrays and "labels/segmentations" arrays and "coordinates" arrays.

Reply:
@adamjstewart if you have experience with this sort of typing approach, we can certainly use the help! 🙏

Reply:
I have a lot of experience with basic typing and almost no experience with complex typing. Not sure how much help I can be, but let me know what you come up with. Should I close this PR or wait until you come up with a final set of suggested type hints for "image" and "mask" arrays?